<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>938</width>
    <height>529</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>JMSK</string>
  </property>
  <property name="windowIcon">
   <iconset resource="jmsk.qrc">
    <normaloff>:/images/primary-modem.svg</normaloff>:/images/primary-modem.svg</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QWidget" name="widget_3" native="true">
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QSplitter" name="splitter">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="childrenCollapsible">
          <bool>false</bool>
         </property>
         <widget class="Console" name="console" native="true"/>
         <widget class="TextInputWidget" name="inputwidget" native="true"/>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="widget_2" native="true">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>140</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>140</height>
       </size>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,0">
       <property name="spacing">
        <number>2</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QSpectrumDisplay" name="spectrumdisplay" native="true">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>130</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QScatterPlot" name="scatterplot" native="true">
         <property name="minimumSize">
          <size>
           <width>130</width>
           <height>130</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>130</width>
           <height>130</height>
          </size>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="widget" native="true">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>30</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>30</height>
       </size>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="spacing">
        <number>7</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLed" name="ledvolume">
         <property name="minimumSize">
          <size>
           <width>25</width>
           <height>25</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>25</width>
           <height>25</height>
          </size>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="pixmap">
          <pixmap resource="jmsk.qrc">:/images/off_led.svg</pixmap>
         </property>
         <property name="scaledContents">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>25</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>25</height>
          </size>
         </property>
         <property name="text">
          <string>Volume</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLed" name="ledsignal">
         <property name="minimumSize">
          <size>
           <width>25</width>
           <height>25</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>25</width>
           <height>25</height>
          </size>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="pixmap">
          <pixmap resource="jmsk.qrc">:/images/off_led.svg</pixmap>
         </property>
         <property name="scaledContents">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_3">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>25</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>25</height>
          </size>
         </property>
         <property name="text">
          <string>Signal</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Minimum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>15</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="label_4">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>25</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>25</height>
          </size>
         </property>
         <property name="text">
          <string>Speed</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBoxbps">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <item>
          <property name="text">
           <string>50 bps</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>125 bps</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>250 bps</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>500 bps</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>600 bps</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>1000 bps</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>1200 bps</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>1225 bps</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_5">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>25</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>25</height>
          </size>
         </property>
         <property name="text">
          <string>Locking</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBoxlbw">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <item>
          <property name="text">
           <string>75 Hz</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>187.5 Hz</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>375 Hz</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>500 Hz</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>750 Hz</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>900 Hz</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>1500 Hz</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>1800 Hz</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>1837.5 Hz</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBoxsql">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <item>
          <property name="text">
           <string>SQL off</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>SQL on</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBoxafc">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <item>
          <property name="text">
           <string>AFC off</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>AFC on</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_6">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>25</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>25</height>
          </size>
         </property>
         <property name="text">
          <string>Display</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBoxdisplay">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <item>
          <property name="text">
           <string>Constellation</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Symbol phase</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>None</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>938</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="action_Exit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="action_About"/>
   </widget>
   <widget class="QMenu" name="menu_Tools">
    <property name="title">
     <string>&amp;Tools</string>
    </property>
    <addaction name="action_Settings"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Tools"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionCleanConsole"/>
   <addaction name="actionConnectToUDPPort"/>
   <addaction name="actionRawOutput"/>
   <addaction name="separator"/>
   <addaction name="action_Settings"/>
   <addaction name="separator"/>
   <addaction name="actionClearTXWindow"/>
   <addaction name="actionIdleTX"/>
   <addaction name="actionBeacon"/>
   <addaction name="actionTXRX"/>
  </widget>
  <action name="action_Exit">
   <property name="icon">
    <iconset resource="jmsk.qrc">
     <normaloff>:/images/application-exit.png</normaloff>:/images/application-exit.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Exit</string>
   </property>
  </action>
  <action name="action_About">
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
  <action name="actionCleanConsole">
   <property name="icon">
    <iconset resource="jmsk.qrc">
     <normaloff>:/images/clear.png</normaloff>:/images/clear.png</iconset>
   </property>
   <property name="text">
    <string>CleanConsole</string>
   </property>
   <property name="toolTip">
    <string>Clear the RX window</string>
   </property>
  </action>
  <action name="actionConnectToUDPPort">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="jmsk.qrc">
     <normaloff>:/images/disconnect.png</normaloff>
     <normalon>:/images/connect.png</normalon>:/images/disconnect.png</iconset>
   </property>
   <property name="text">
    <string>ConnectToUDPPort</string>
   </property>
   <property name="toolTip">
    <string>Connects RX bit stream to the UDP port instead of the RX window</string>
   </property>
  </action>
  <action name="actionRawOutput">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="jmsk.qrc">
     <normaloff>:/images/primary-binary.svg</normaloff>:/images/primary-binary.svg</iconset>
   </property>
   <property name="text">
    <string>RawOutput</string>
   </property>
   <property name="toolTip">
    <string>Outputs raw data to UDP port</string>
   </property>
  </action>
  <action name="actionTXRX">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="jmsk.qrc">
     <normaloff>:/images/rx.svg</normaloff>
     <normalon>:/images/tx.svg</normalon>
     <activeon>:/images/tx.svg</activeon>
     <selectedon>:/images/tx.svg</selectedon>:/images/rx.svg</iconset>
   </property>
   <property name="text">
    <string>TX</string>
   </property>
   <property name="toolTip">
    <string>Toggle transmit/receive</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actionIdleTX">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="jmsk.qrc">
     <normaloff>:/images/eighth note.svg</normaloff>:/images/eighth note.svg</iconset>
   </property>
   <property name="text">
    <string>IdleTX</string>
   </property>
   <property name="toolTip">
    <string>Continues to transmit when idle</string>
   </property>
  </action>
  <action name="actionClearTXWindow">
   <property name="icon">
    <iconset resource="jmsk.qrc">
     <normaloff>:/images/clear.png</normaloff>:/images/clear.png</iconset>
   </property>
   <property name="text">
    <string>clearTXWindow</string>
   </property>
   <property name="toolTip">
    <string>Clear the TX window</string>
   </property>
  </action>
  <action name="action_Settings">
   <property name="icon">
    <iconset resource="jmsk.qrc">
     <normaloff>:/images/settings.png</normaloff>:/images/settings.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Settings</string>
   </property>
   <property name="toolTip">
    <string>Open settings dialog</string>
   </property>
  </action>
  <action name="actionBeacon">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="jmsk.qrc">
     <normaloff>:/images/beacon.svg</normaloff>:/images/beacon.svg</iconset>
   </property>
   <property name="text">
    <string>Beacon</string>
   </property>
   <property name="toolTip">
    <string>Enable TX beacon</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Console</class>
   <extends>QWidget</extends>
   <header>gui_classes/console.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QScatterPlot</class>
   <extends>QWidget</extends>
   <header>gui_classes/qscatterplot.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QSpectrumDisplay</class>
   <extends>QWidget</extends>
   <header>gui_classes/qspectrumdisplay.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QLed</class>
   <extends>QLabel</extends>
   <header>gui_classes/qled.h</header>
  </customwidget>
  <customwidget>
   <class>TextInputWidget</class>
   <extends>QWidget</extends>
   <header>gui_classes/textinputwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="jmsk.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>action_Exit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>428</x>
     <y>243</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
